test/test_gmm.py

```diff
@@ -409,8 +255,9 @@ def test_gmm_backward(self):
             group_sizes.to("xla"))
         # same gmm/tgmm was run for the `test_cache=False` case so the
         # cache should be populated now
+        new_cnt = xr.get_num_cached_compilation_graph()
         if test_cache:
-          self.assertEqual(met.counter_value('trace_pallas_cache_hit'), 2)
+          self.assertEqual(new_cnt, old_cnt)
```

Comment:
based on the previous test, shouldn't you also need to test the case when test_cache=False: new_cnt==old_cnt+2?

Reply:
I think the original test code is to make sure that there's no recompilation, the new code did the same thing.